{
    "body": "### Steps to reproduce the issue\r\n\r\n```console\r\n$ spack install autoconf arch=linux-rhel7-power8le\r\n==> Warning: Using GCC 4.8 to optimize for Power 8 might not work if you are not on Red Hat Enterprise Linux 7, where a custom backport of the feature has been done. Upstream support from GCC starts in version 4.9\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/libsigsegv-2.12-txb7lxpsvmwony66zbsjdvp6pvcqgdmi\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/berkeley-db-18.1.40-t3x7wr3zfa6qkvf64wvvgubnltu2tvyl\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/pkgconf-1.7.3-vzmd2eiszaid2mz7cdyufa6a5wtgstmp\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/m4-1.4.18-ujfdwtaemutf3es2doezt4x6rfxiaanm\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/ncurses-6.2-uszrlwxzdwe3nyj2hwhb3jkhtc5lgpkv\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/readline-8.0-djztk7gruh65ryoa4zrvpyiz5rnnmrhz\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/gdbm-1.18.1-2qarpshhhmcc3fozbljikl26t6f5gpls\r\n[+] /home/user/spack-0.16.2/opt/spack/linux-rhel7-power8le/gcc-4.8.5/perl-5.32.0-yegtja5ecg5bb36mieemj3q2imtd527l\r\n==> Installing autoconf-2.69-nhgcmyb562pjbsaiie2fokkdielckwwm\r\n==> No binary for autoconf-2.69-nhgcmyb562pjbsaiie2fokkdielckwwm found: installing from source\r\n==> Using cached archive: /home/user/spack-0.16.2/var/spack/cache/_source-cache/archive/95/954bd69b391edc12d6a4a51a2dd1476543da5c6bbf05a95b59dc0dd6fd4c2969.tar.gz\r\n==> autoconf: Executing phase: 'autoreconf'\r\n==> [2021-06-10-12:20:41.891010] CalledProcessError: Command '['/tmp/netuser/spack-stage/spack-stage-autoconf-2.69-nhgcmyb562pjbsaiie2fokkdielckwwm/spack-src/build-aux/config.guess']' returned non-zero exit status 1.\r\n==> Error: RuntimeError: Failed to find suitable substitutes for config.guess\r\n\r\n/home/user/spack-0.16.2/lib/spack/spack/build_systems/autotools.py:170, in _do_patch_config_files:\r\n        167\r\n        168        # Check that we found everything we needed\r\n        169        if to_be_found:\r\n  >>    170            msg = 'Failed to find suitable substitutes for {0}'\r\n        171            raise RuntimeError(msg.format(', '.join(to_be_found)))\r\n        172\r\n        173        # Copy the good files over the bad ones\r\n```\r\n\r\n### Information on your system\r\n\r\n```\r\nspack debug report\r\n* **Spack:** 0.16.2\r\n* **Python:** 3.6.8\r\n* **Platform:** linux-rhel7-power8le\r\n```\r\n\r\n### Additional information\r\n\r\nThis is the same error as in https://github.com/spack/spack/pull/21256, but here the file config.guess (and config.sub) is executable:\r\n```\r\n$ tar -tvf autoconf-2.69.tar.gz |grep config\r\n-rwxrwxr-x 0/0           35516 2012-04-24 19:37 autoconf-2.69/build-aux/config.sub\r\n-rwxrwxr-x 0/0           44826 2012-04-24 19:15 autoconf-2.69/build-aux/config.guess\r\n```\r\n\r\nSame error is reported for autoconf 2.62 and 2.59.\r\nAutoconf 2.13 instead is so old that one of the patches fails with:\r\n```\r\n==> Error: FileNotFoundError: [Errno 2] No such file or directory: 'bin/autom4te.in'\r\n\r\n/home/user/spack-0.16.2/var/spack/repos/builtin/packages/autoconf/package.py:43, in patch:\r\n         40        # because autom4te is called during the build\r\n         41        filter_file('^#! @PERL@ -w',\r\n         42                    '#! /usr/bin/env perl',\r\n  >>     43                    'bin/autom4te.in')\r\n```\r\n\r\n* `spack-build-out.txt` only contains a `CalledProcessError: Command '['/tmp/netuser/spack-stage/spack-stage-autoconf-2.69-nhgcmyb562pjbsaiie2fokkdielckwwm/spack-src/build-aux/config.guess']' returned non-zero exit status 1.`\r\n* [spack-build-env.txt](https://github.com/spack/spack/files/6633907/spack-build-env.txt)\r\n\r\n### General information\r\n\r\n- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform\r\n- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers\r\n- [x] I have uploaded the build log and environment files\r\n- [x] I have searched the issues of this repo and believe this is not a duplicate",
    "user": "bebosudo",
    "url": "https://api.github.com/repos/spack/spack/issues/24248",
    "updated_at": "2021-09-27 22:38:15",
    "created_at": "2021-06-10 19:33:43",
    "closed_at": "2021-09-27 22:38:15",
    "state": "closed",
    "title": "Installation issue: autoconf on ppc64le",
    "number": 24248,
    "milestone": null,
    "labels": [
        "build-error"
    ],
    "id": 917787621,
    "html_url": "https://github.com/spack/spack/issues/24248",
    "assignees": [],
    "comments": 1
}